ModComponent/Mapper/ZipFileLoader.cs

```diff
@@ -198,12 +189,19 @@ private static bool TryHandleJson(string zipFilePath, string internalPath, strin
 			else if (internalPath.StartsWith(@"blueprints/"))
 			{
 				Logger.Log($"Reading blueprint json from zip at '{internalPath}'");
-				CraftingRevisions.BlueprintManager.AddBlueprintFromJson(text, false);
+				CraftingRevisions.BlueprintManager.AddBlueprintFromJson(text);
 			}
 			else if (internalPath.StartsWith(@"localizations/"))
 			{
-				Logger.Log($"Reading json localization from zip at '{internalPath}'");
-				LocalizationUtilities.LocalizationManager.LoadJSONLocalization(text);
+				// change emthod to ensure we go via the BOM fixed methods..
+				TextAsset textAsset = new(text);
+				LocalizationUtilities.LocalizationManager.LoadLocalization(textAsset, internalPath);
```

Comment:
Changing to TextAsset is not an acceptable solution. Find another way.

Reply:
Can you elaborate why this isn't acceptable ?

Reply:
The only reason TextAssets are accepted in LoadLocalization is to reduce friction with loading text files from asset bundles.

Also, `text` should not have BOM at this point. If it does, that is a huge problem.

Reply:
Restored & Implemented same BOM check as LocalizationUtils